src/parser/jobs/whm/modules/Lilies.tsx

```diff
@@ -0,0 +1,167 @@
+import {Plural, Trans} from '@lingui/react'
+import Color from 'color'
+import {DataLink} from 'components/ui/DbLink'
+import {Event} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge, Gauge as CoreGauge, TimerGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const LILY_INITIAL_STACKS = 0
+const LILY_MAX_STACKS = 3
+const LILY_60_TIME_REQUIRED = 30000
+const LILY_TIME_REQUIRED = 20000
+
+const BLOOD_LILY_BLOOM = 3
+const MISERY_COST = 3
+
+const GAUGE_FADE = 0.25
+const TIMER_FADE = 0.75
+const BLOODLILY_FADE = 0.5
+const LILY_COLOR = Color('#4f73b5')
+const BLOODLILY_COLOR = Color('#b52d6c')
+
+export class Lilies extends CoreGauge {
+
+	@dependency private suggestions!: Suggestions
+
+	private lilyConsumers = [
+		this.data.actions.AFFLATUS_RAPTURE.id,
+		this.data.actions.AFFLATUS_SOLACE.id,
+	]
+
+	private bloodLilyConsumers = [
+		this.data.actions.AFFLATUS_MISERY.id,
+	]
```

Comment:
These are single use, not worth defining inline. Instead, you can have a const with the ActionKeys, and then use those in the hooks like this example:
```ts
// const
const LILY_CONSUMERS: ActionKey[] = [
  'AFFLATUS_RAPTURE',
  'AFFLATUS_SOLUS',
]

// hook
this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(LILY_CONSUMERS)), this.onLilySpend)
```